### ISSUE TITLE
`process.env.VAR` being split into multiple lines breaks `rollup-plugin-replace`

### ISSUE THREAD

**Issue by ansballard** [prettier/prettier#3048]
**Prettier 1.7.4**
[Playground link](https://prettier.io/playground/#%7B%22content%22%3A%22function%20testing()%20%7B%5Cn%20%20const%20p%20%3D%20%7B%7D%3B%5Cn%20%20%2F%2F%20faking%20some%20tabs%20since%20I%20can't%20paste%20my%20real%20code%20in%5Cn%20%20if(true)%20%7B%5Cn%20%20%20%20if(false)%20%7B%5Cn%20%20%20%20%20%20return%20%60%24%7Bprocess.env.OPENID_URL%7D%2Fsomething%2Fsomething%2Fsomething%3F%24%7BObject.keys(p)%5Cn%20%20%20%20%20%20%20%20.map(k%20%3D%3E%20%60%24%7BencodeURIComponent(k)%7D%3D%24%7BencodeURIComponent(p%5Bk%5D)%7D%60)%5Cn%20%20%20%20%20%20%20%20.join(%5C%22%26%5C%22)%7D%60%3B%5Cn%20%20%20%20%7D%5Cn%20%20%7D%5Cn%7D%22%2C%22options%22%3A%7B%22ast%22%3Afalse%2C%22bracketSpacing%22%3Atrue%2C%22doc%22%3Afalse%2C%22jsxBracketSameLine%22%3Afalse%2C%22output2%22%3Afalse%2C%22parser%22%3A%22babylon%22%2C%22printWidth%22%3A80%2C%22semi%22%3Atrue%2C%22singleQuote%22%3Afalse%2C%22tabWidth%22%3A2%2C%22trailingComma%22%3A%22none%22%2C%22useTabs%22%3Afalse%7D%7D)
```sh
**Input:**
```jsx
function testing() {
  const p = {};
  // faking some tabs since I can't paste my real code in
  if(true) {
    if(false) {
      return `${process.env.OPENID_URL}/something/something/something?${Object.keys(p)
        .map(k => `${encodeURIComponent(k)}=${encodeURIComponent(p[k])}`)
        .join("&")}`;
    }
  }
}
```

**Output:**
```jsx
function testing() {
  const p = {};
  // faking some tabs since I can't paste my real code in
  if (true) {
    if (false) {
      return `${process.env
        .OPENID_URL}/something/something/something?${Object.keys(p)
        .map(k => `${encodeURIComponent(k)}=${encodeURIComponent(p[k])}`)
        .join("&")}`;
    }
  }
}
```
So as far as I could tell, there aren't any issues for this specific case in this or the [rollup plugin repo](https://github.com/rollup/rollup-plugin-replace). The standard way to inject process.env variables using rollup is to use the replace plugin. This rolls through your code and replaces strings that exactly match what you specify. This means if prettier turns `process.env.VAR` into `process.env.\nVAR`, then that environment variable is never replaced, and you get "process is not defined" or whatever other runtime error. I haven't run into this with the webpack.DefinePlugin, but I assume it works the same way, and would break as well.

So I'm not sure if this even needs to get fixed in this repo, or if maybe the plugins need to account for whitespace/be smarter in general about how they inject, but it's defintely an issue, and I figured this was the best place to start a discussion.

Thanks, let me know if there's any more info I can provide!

**Comment by lydell** [comment]
I think it might be reasonable to add a special case in Prettier for this. As a workaround, you can do `const openIdUrl = process.env.OPENID_URL` above the template literal.

**Comment by suchipi** [comment]
I think this is worth special-casing, too. In an ideal world, tools would look for process.env.FOO via the AST, but preserving it results in a better DX all-around (it's more greppable too)

**Comment by azz** [comment]
In general don't we want it to look something more like:


```js
      return `${
        process.env.OPENID_URL
      }/something/something/something?${
        Object.keys(p)
          .map(k => `${encodeURIComponent(k)}=${encodeURIComponent(p[k])}`)
          .join("&")
      }`;
    }
  }
```

?

**Comment by SimenB** [comment]
The last one is https://github.com/prettier/prettier/issues/1893

I'd love to see that one fixed, chains in template strings is the only thing on my personal "stuff prettier makes way worse" list

**Comment by duailibe** [comment]
This was fixed by #3124
